Add more learner-friendly explanation and missing test for ambiguous column error

danielhollas opened the pull request:
### High-level description of change

This PR is a followup on #717, adding a more detailed explanation to a rather cryptic ambiguous column error. I also added a missing test.

### Are there performance implications for this change?

No.

### Have you added tests to cover this new/updated code?

I added a missing test.

### Risks involved
To avoid deleting existing translations, I added the new message as a separate string.
It looks a bit ugly in the code but is translator-friendly (I checked that more than 8 lang translated the error already). But LMK if you want to change it, it's not a deal-breaker.

### Are there any dependencies or blockers for merging this?

No.

### How can we verify that this change works?

The following code triggers the error at http://localhost:8000/demos/simple/sql.html
```sql
CREATE TABLE students (id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT, grade_year INTEGER NOT NULL);
CREATE TABLE student_grades (id INTEGER, grade INTEGER);
INSERT INTO students VALUES (1, "Brian", 3);
INSERT INTO student_grades VALUES (1, 95);
SELECT id FROM students INNER JOIN student_grades on students.id = student_grades.id;
```
![image](https://user-images.githubusercontent.com/9539441/78512354-10e4e680-77a4-11ea-85e6-03f540ce1039.png)

pamelafox commented on js/output/sql/sql-output.js:
```diff
@@ -102,6 +102,14 @@ window.SQLOutput = Backbone.View.extend({
+            // I18N: Keep the space at the beginning of string.
+            errorMessage += " " + i18n._("Multiple tables that you're joining " +
+               "contain a column with that name. To use that column in your query, " +
+               "specify the table of the column. Example: firstTable.\"%(colName)s\"",
```
Hm, seeing it in the screenshot, the example is throwing me off. It's weird for me to see a generic table name (firstTable) and then a specific column name (id). It's halfway between being an example and being the real thing.
 
Another idea is this sentence instead of "To use..."
"Specify the name of the table that contains the column, using the format "tableName.columnName".
(I'm not sure we need "to use that column in your query", since the error implies they are trying to use it, right?")
What do you think? What's most helpful?

danielhollas replied:
Agreed, I was not very satisfied either. I'll change it tomorrow. We do have access to the SQL statement that throws the error so we could try to extract at least one table name with some clever regexing. LMK if you think I should pursue that.

pamelafox replied:
I don't think it's worth it, since it's possible they mean the other table, right?

danielhollas replied:
@pamelafox right, good point, that would actually perhaps make it more confusing. I updated the wording and rebased on master, see comment below. Thanks!

pamelafox commented on js/output/sql/sql-output.js:
```diff
@@ -102,6 +102,14 @@ window.SQLOutput = Backbone.View.extend({
+            // to preserve existing translations.
+            // I18N: This Oh Noes message follows "Ambiguous column name" SQL error
+            // I18N: Keep the space at the beginning of string.
+            errorMessage += " " + i18n._("Multiple tables that you're joining " +
```
I'm fine with leaving "Ambiguous column name" as a prefix to the error.

danielhollas commented:
Updated wording (and rebased on master)

![image](https://user-images.githubusercontent.com/9539441/78811888-da3ce500-79ca-11ea-97b0-10c097627ddb.png)

Another option how to help student would be to mention this in the SQL documentation, perhaps for [inner join and outer join](https://www.khanacademy.org/computer-programming/sql-join-on-tables/5409956539006976).

danielhollas commented:
@pamelafox I've also fixed the SQL demo page, Oh Noes was unhappy about "char" and "int".

![image](https://user-images.githubusercontent.com/9539441/78813639-9bf4f500-79cd-11ea-9c75-1b073a1cc63b.png)

pamelafox commented:
I think the talkie does mention it, per this transcript screenshot?

<img width="715" alt="Screen Shot 2020-04-08 at 3 45 24 PM" src="https://user-images.githubusercontent.com/297042/78840578-fed79380-79af-11ea-8102-15eb313fbe9a.png">
